src/west/commands/project.py

```diff
@@ -754,15 +753,40 @@ def _expand_shorthands(project, s):
     # Expands project-related shorthands in 's' to their values,
     # returning the expanded string
 
+    # Some of the trickier ones below. 'qual' stands for 'qualified', meaning
+    # the full path to the ref (e.g. refs/heads/master).
+    #
+    # qual-rev:
+    #   The qualified local branch for the revision from the manifest (e.g.
+    #   refs/heads/master), or an SHA if the revision is an SHA
+    #
+    # qual-remote-rev:
+    #   The qualified remote branch for the revision (e.g.
+    #   remotes/origin/master), or an SHA if the revision is an SHA
+    #
+    # manifest-rev-branch:
+    #   The name of the magic branch that points to the manifest revision
+    #
+    # qual-manifest-rev-branch:
+    #   A qualified reference to the magic manifest revision branch, e.g.
+    #   refs/heads/manifest-rev
+
     return s.replace('(name)', project.name) \
             .replace('(name-and-path)',
                      '{} ({})'.format(
                          project.name, os.path.join(project.path, ""))) \
             .replace('(url)', project.url) \
             .replace('(path)', project.path) \
             .replace('(abspath)', project.abspath) \
-            .replace('(revision)', project.revision) \
+            .replace('(qual-rev)',
+                     project.revision if _is_sha(project.revision) else \
+                     'refs/heads/' + project.revision) \
+            .replace('(qual-remote-rev)',
+                     project.revision if _is_sha(project.revision) else \
+                     'remotes/origin/' + project.revision) \
```

Comment:
Yup, hardcoded to `origin` because it hasn't been made configurable yet.

Reply:
Note: this will not be "configurable" but rather: "equal to the `remote` field in the manifest" as per https://github.com/zephyrproject-rtos/west/issues/73#issuecomment-429002132